client/state/sites/selectors.js

```diff
@@ -1020,3 +1020,19 @@ export const hasDefaultSiteTitle = ( state, siteId ) => {
 	// we are using startsWith here, as getSiteSlug returns "slug.wordpress.com"
 	return site.name === i18n.translate( 'Site Title' ) || startsWith( slug, site.name );
 };
+
+/**
+ * Determines if site's permalinks are editable
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site's permalinks are editable
+ */
+export function areSitePermalinksEditable( state, siteId ) {
+	const permalinkStructure = getSiteOption( state, siteId, 'permalink_structure' );
+	if ( ! permalinkStructure ) {
+		return false;
+	}
+
+	return /\/\%postname\%\/?/.test( permalinkStructure );
```

Comment:
Guessing this is partly legacy, but I don't think we need to escape `%` in the RegExp.